### PR TITLE
[keymap]  ergodash keymap for yet-another-developer

### PR DESCRIPTION
**Pull request by yet-another-developer**
## Description

This adds an ergodash keymap for user yet-another-developer as well as his own userspace.
This follows the code patterns from keymap/s and userspace/s of @drashna and @kuchosauronad0.
This contributes a template for ergodash owners to have the features and best practices laid out by @drashna and @kuchosauronad0.

The keymap layout was prettified to mirror the physical layout of the keys as opposed to the default ergodash keymap layout

## Types of Changes

- [x] Keymap/layout/userspace (addition or update)

## Checklist

- [x] My code follows the code style of this project.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: users/yet-another-developer/process_records.c [hunk 1]
```diff
@@ -0,0 +1,201 @@
...
+                    #if defined(__arm__)
+                    send_string_with_delay_P(PSTR(":dfu-util"), TAP_CODE_DELAY);
+                    #elif defined(BOOTLOADER_DFU)
+                    send_string_with_delay_P(PSTR(":dfu"), TAP_CODE_DELAY);
+                    #elif defined(BOOTLOADER_HALFKAY)
+                    send_string_with_delay_P(PSTR(":teensy"), TAP_CODE_DELAY);
+                    #elif defined(BOOTLOADER_CATERINA)
+                    send_string_with_delay_P(PSTR(":avrdude"), TAP_CODE_DELAY);
+                    #endif // bootloader options
+                }
```
Comment (drashna) [inline users/yet-another-developer/process_records.c]: This entire block can de replaced now.

All you need is:
```c
                    send_string_with_delay_P(PSTR(":flash"), TAP_CODE_DELAY);
```

This takes care of the detection.

File: users/yet-another-developer/leader.c [hunk 2]
```diff
@@ -0,0 +1,55 @@
+#include "leader.h"
+
+LEADER_EXTERNS();
+
+#define TAP_ONCE(code)  \
+  register_code (code); \
+  unregister_code (code)
+
```
Comment (drashna) [inline users/yet-another-developer/leader.c]: Can be removed, since `tap_code` does this.

File: users/yet-another-developer/leader.c [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+// Runs constantly in the background, in a loop.
+void matrix_scan_user(void){
+  static bool has_ran_yet;
+  if (!has_ran_yet) {
+    has_ran_yet = true;
+    startup_user();
```
Comment (drashna) [inline users/yet-another-developer/leader.c]: Not needed anymore.   You can use `keyboard_post_init_user` now.

Comment (yet-another-developer) [reply users/yet-another-developer/leader.c]: Is it safe to remove from `matrix_scan_user`? I checked your userspace `matrix_scan_user` and it still has the same code in `drashna.c`

Comment (yet-another-developer) [reply users/yet-another-developer/leader.c]: I've removed the code regardless, kindly review if changes are faultless. 
I've removed the code for both {user}.c and in leader.c under `matrix_scan_user`.

Comment (drashna) [reply users/yet-another-developer/leader.c]: It does?  Whoops, need to remove that too, then! :)

**Pull request merged**